pkgs/os-specific/linux/libbpf/default.nix

```diff
@@ -1,37 +1,28 @@
-{ fetchFromGitHub
-, fetchpatch
+{ kernel
 , libelf
 , pkg-config
+, python3
 , stdenv
 , zlib
 , lib
 , nixosTests
 }:
 
 stdenv.mkDerivation rec {
-  pname = "libbpf";
-  version = "0.7.0";
+  inherit (kernel) src version;
 
-  src = fetchFromGitHub {
-    owner = "libbpf";
-    repo = "libbpf";
-    rev = "v${version}";
-    sha256 = "sha256-NFVJ8JquWVzu+QoaaOMzhnu6/IqdP1FPhtJFidXA4L4=";
-  };
+  pname = "libbpf";
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config python3 ];
   buildInputs = [ libelf zlib ];
 
   enableParallelBuilding = true;
-  makeFlags = [ "PREFIX=$(out)" "-C src" ];
+  makeFlags = [ "prefix=$(out)" ];
 
-  passthru.tests = {
-    bpf = nixosTests.bpf;
-  };
+  passthru.tests = { inherit (nixosTests) bpf; };
 
-  postInstall = ''
-    # install linux's libbpf-compatible linux/btf.h
```

Comment:
It seems to me that if you want to use libbpf, you also must use the linux headers for your `linuxPackages`. I think perhaps it might be useful to make `linuxHeaders` a propagated input. That might mean making the headers a proper `linuxPackage` as well, but I am unsure if there is a good reason it is not, already.

Reply:
It would mean we'd have to rebuild every package that used the headers for every kernel version!  It's pretty much never a problem to use a newer version of the headers, so `linuxHeaders` is usually the latest version of Linux.